python/paddle/fluid/optimizer.py

```diff
@@ -5019,16 +5019,13 @@ def _insert_loss_scale(self, block):
         if self._num_microbatches == 1: return
         for index, op in reversed(tuple(enumerate(list(block.ops)))):
             if self._is_loss_grad_op(op):
-                loss_grad_var = block.vars[op.output_arg_names[0]]
-                block._insert_op(
-                    index=index + 1,
-                    type='scale',
-                    inputs={'X': loss_grad_var},
-                    outputs={'Out': loss_grad_var},
-                    attrs={
-                        'scale': 1.0 / self._num_microbatches,
-                        self._op_role_key: self._op_role.Backward
-                    })
+                assert op.type == 'fill_constant', \
+                    "loss_grad_op must be fill_constant op, " \
+                    "but this op is {}".format(op.type)
+                assert op.has_attr('value')
+                loss_scale = float(op.attr('value'))
```

Comment:
Mind the potential precision loss here. fill_constant op will cast the value into fp32 and then save  as string into its OpDesc. reload & reset this value might cause precision loss when the denominator is odd （3，7， 11， etc）

Reply:
![image](https://user-images.githubusercontent.com/10208305/133542003-965b6bbc-7482-4b7f-9fe0-0191d18bd2df.png)
![image](https://user-images.githubusercontent.com/10208305/133542409-ffb1d679-f99f-44b0-900f-a20e28df6a65.png)
loss_grad_op(that is fill_constant) use `value` instead of `str_value`, which AttrType is `float`. `value` will be saved with `float32` in protobuf. If we encount precision problem, I this this must be caused by `float` AttrType, and I think `double` AttrType is better, which framework does not provide.

Reply:
![image](https://user-images.githubusercontent.com/10208305/133544198-b0813f95-8ca2-4f7b-add1-639320977b3b.png)
the type of op.attr('value') is already `float64` in python, add float(op.attr('value')) is only for explicit.